cli/cmd/resources/instrumentor.go

```diff
@@ -68,6 +68,54 @@ func NewInstrumentorRoleBinding(ns string) *rbacv1.RoleBinding {
 	}
 }
 
+func NewInstrumentorRoleBinding(ns string) *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RoleBinding",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "odigos-instrumentor",
+			Namespace: ns,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind: "ServiceAccount",
+				Name: "odigos-instrumentor",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     "odigos-instrumentor",
+		},
+	}
+}
+
+func NewInstrumentorRole(ns string) *rbacv1.Role {
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Role",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "odigos-instrumentor",
+			Namespace: ns,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs: []string{
+					"get",
+				},
+				APIGroups: []string{""},
+				Resources: []string{
+					"secrets",
```

Comment:
I think it's good for now.
In the future, we might want to consider storing the `tier` in the `odigos-deployment` configmap, so we can read the tier easily without needing permissions to secrets. We can also use `envFrom` to automatically have these as env variables in the code